lucille-core/src/main/java/com/kmwllc/lucille/indexer/SolrDocRequests.java

```diff
@@ -3,32 +3,35 @@
 import com.kmwllc.lucille.core.Document;
 import org.apache.solr.common.SolrInputDocument;
 
-import java.util.ArrayList;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
+import java.util.*;
 
 /**
  * Maintains the list of Solr Add and Update requests that are built up as a SolrIndexer processes a
  * batch. Allows the calling code (typically a SolrIndexer instance) to keep track of the document
- * ids within a batch and check for collisions.
+ * ids within a batch and check for collisions. Maintains separate list of documents to delete
+ * based on the configured delete marker on incoming documents, these are held in a list of ids for documents
+ * that can be deleted directly by the ID or a map of field names to field values for documents that have to
+ * be deleted with a terms query.
  */
 public class SolrDocRequests {
   private final List<SolrInputDocument> docsToAddOrUpdate;
   private final Set<String> idsToAddOrUpdate;
   private final Set<String> idsToDelete;
+  private final Map<String,List<String>> valuesToDeleteByField;
 
   public SolrDocRequests() {
-    this(new ArrayList<>(), new HashSet<>(), new HashSet<>());
+    this(new ArrayList<>(), new HashSet<>(), new HashSet<>(), new HashMap<>());
```

Comment:
Was going to suggest to initialize these as null and lazily instantiate upon the first add. Looking into it... storage for ArrayList and HashMap is initialized on first use, so there wouldn't be much savings here (basically just the pointers). No action being requested in this comment.